Bottom: a500657e3404d72a3a8e50dde9f60a5e7a25443d
Top:    8287cc82c4998eb3744454cbcc00f1c20e87b3bd
Author: Jason Wang <jasowang@redhat.com>
Date:   2018-02-07 20:54:03 +0800

vhost: packed ring support

Signed-off-by: Jason Wang <jasowang@redhat.com>


---

diff --git a/drivers/vhost/vhost.h b/drivers/vhost/vhost.h
index 7876a3d..519d9df 100644
--- a/drivers/vhost/vhost.h
+++ b/drivers/vhost/vhost.h
@@ -91,7 +91,10 @@ struct vhost_virtqueue {
 	/* The actual ring of buffers. */
 	struct mutex mutex;
 	unsigned int num;
-	struct vring_desc __user *desc;
+	union {
+		struct vring_desc __user *desc;
+		struct vring_packed __user *packed;
+	};
 	struct vring_avail __user *avail;
 	struct vring_used __user *used;
 	const struct vhost_umem_node *meta_iotlb[VHOST_NUM_ADDRS];
